tgui/packages/tgui/interfaces/Sensors.tsx

```diff
@@ -0,0 +1,598 @@
+  id_status: string;
+  id_class: string;
+  id_name: string;
+  can_change_class: BooleanLike;
+  can_change_name: BooleanLike;
+  contact_details: string;
+
+  distress_beacons: DistressBeaconData[];
+  distress_range: number;
+
+  datalink_requests: { name: string; ref: string }[];
+  datalinked: { name: string; ref: string }[];
+};
+
+type ContactData = {
+  name: string;
+  ref: string;
+  bearing: number;
+  can_datalink: BooleanLike;
+  landed: BooleanLike;
+  distance: number;
+  x: number;
+  y: number;
+  color: string; // assigned in ts
+};
+
+type DistressBeaconData = {
+  caller: string;
+  sender: string;
+  bearing: number;
+};
+
+const SensorSection = function (act, data: SensorsData) {
+  const range_choice_max = data.range_choices[data.range_choices.length - 1];
+
+  return (
+    <Section
+      title="Sensor Array Control"
+      buttons={
+        <Button
+          content={
+            'Sector Map View ' + (data.viewing ? 'Engaged' : 'Disengaged')
+          }
+          onClick={() => act('viewing')}
+        />
+      }>
+      <Table>
+        <Table.Row>
+          <Table.Cell>State:</Table.Cell>
+          <Table.Cell>
+            {data.on ? 'Active' : 'Inactive'}
+            {', '}
+            <Button content="on/off" onClick={() => act('toggle')} />
+          </Table.Cell>
+        </Table.Row>
+        <Table.Row>
+          <Table.Cell>Status:</Table.Cell>
+          <Table.Cell>{data.status}</Table.Cell>
+        </Table.Row>
+        <Table.Row>
+          <Table.Cell>Range:</Table.Cell>
+          <Table.Cell>
+            <Slider
+              animated
+              step={1}
+              stepPixelSize={16}
+              value={data.desired_range}
+              minValue={1}
+              maxValue={range_choice_max}
+              onChange={(e, value) =>
+                act('range_choice', { range_choice: value })
+              }>
+              Desired Range: {data.desired_range} / {range_choice_max}
+            </Slider>
+            <ProgressBar
+              animated
+              backgroundColor={Color.lerp(
+                new Color(62, 97, 137, 0),
+                new Color(189, 32, 32),
+                data.range / range_choice_max
+              )}
+              minValue={1}
+              maxValue={range_choice_max}
+              value={data.range}>
+              Current Range: {data.range} / {range_choice_max}
+            </ProgressBar>
+          </Table.Cell>
+        </Table.Row>
+        {data.deep_scan_range > 0 ? (
+          <Table.Row>
+            <Table.Cell>{data.deep_scan_name}:</Table.Cell>
+            <Table.Cell>
+              Effective Range: {data.deep_scan_range}
+              {', '}
+              {data.deep_scan_toggled ? 'Active' : 'Inactive'}
+              {', '}
+              <Button
+                content={'on/off'}
+                onClick={() =>
+                  act('deep_scan_toggle', { deep_scan_toggle: true })
+                }
```

Comment:
Same as before, just make it one button.

Reply:
done